### PR TITLE
Add more options and operations to client

### PR DESCRIPTION
**Pull request by sijie**
- Get and Delete Ops
- Complete the Put, Get and Delete options
- More comments

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/options/DeleteOption.java [hunk 1]
```diff
@@ -8,18 +11,63 @@ public static Builder newBuilder() {
...
+         * Set the end key of the delete request. If it is set, the
+         * delete request will delete the keys from <i>key</i> to <i>endKey</i> (exclusive).
+         * <p>If end key is '\0', the range is all keys >= key.
+         * <p>If the end key is one bit larger than the given key, then it deletes all keys with the prefix (the given key).
+         * <p>If both key and end key are '\0', it deletes all keys.
+         *
+         * @param endKey end key
+         * @return builder
+         */
+        public Builder withEndKey(ByteString endKey) {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/options/DeleteOption.java]: withRange(ByteString endKey)?

File: src/main/java/com/coreos/jetcd/options/GetOption.java [hunk 2]
```diff
@@ -1,39 +1,208 @@
...
+         * Set the end key of the get request. If it is set, the
+         * get request will return the keys from <i>key</i> to <i>endKey</i> (exclusive).
+         * <p>If end key is '\0', the range is all keys >= key.
+         * <p>If the end key is one bit larger than the given key, then it gets all keys with the prefix (the given key).
+         * <p>If both key and end key are '\0', it returns all keys.
+         *
+         * @param endKey end key
+         * @return builder
+         */
+        public Builder withEndKey(ByteString endKey) {
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/options/GetOption.java]: withRange?

**Comment by xiang90** [comment]
@sijie LGTM after fixing the name. Thanks!

**Comment by sijie** [comment]
Yup. Changed 'withEndKey' to 'withRange'

**Comment by xiang90** [comment]
lgtm.

**Pull request merged**